pandas/io/json/_json.py

```diff
@@ -850,6 +850,26 @@ def _convert_axes(self):
         obj = self.obj
         assert obj is not None  # for mypy
         for axis_name in obj._AXIS_ORDERS:
+            if (
```

Comment:
This `if` condition seems too specific. Is there a more generic way to handle this?

Reply:
As far as i understood the minimum preconditions to keep string index should in read_json  orient equal to split or index and the index of json format to be a list of int or float strings. Am i right? Should i move the checks in _try_convert_data?
Do you have a more conrete and robust suggestion?

Reply:
I tried a slightly different way. Could you please re-review?